### ISSUE TITLE
how to add new filters?

### ISSUE THREAD

**Issue by macrozone** [unchainedshop/unchained#554]
How can i add a custom filter which i can pass as key-value to  `filterQuery` ?

i have a custom meta data on a product and i want to add a filter that filters by that. how can i do that?

there is a `transformProductSelector` function, but its not called (or only with undefined key and value), so i assume it uses some caching. I don't see anything in the [docs](https://docs.unchained.shop/config/filters/) how to update this cache for all existing products.

**Comment by Mikearaya** [comment]
@macrozone didn't you find it useful, https://docs.unchained.shop/advanced-config/filter/

**Comment by macrozone** [comment]
> @macrozone didn't you find it useful, https://docs.unchained.shop/advanced-config/filter/

unfortunatly not, i extended `transformProductSelector` to include my new key, but this function is never called with any key or value, maybe its cached?

@pozylon do you have an idea?

**Comment by pozylon** [comment]
@macrozone Yes it's cached, the cache is invalidated for products and categories that are changed and on every start/restart of the app unless you have set "skipInvalidationOnStartup" to true

**Comment by macrozone** [comment]
> skipInvalidationOnStartup

that was the missing piece.

would be be good to have a command for that instead ideally.

**Comment by macrozone** [comment]
additional information: i created the custom filter as well through the unchained control panel

**Comment by pozylon** [comment]
If you want to enable a global filter, you can implement `transformFilterSelector` and return additional _ids in the filter selector. this is the default implementation:

```javascript
const defaultSelector = ({ filterIds, filterQuery, includeInactive }: SearchQuery) => {
  const selector: Query = {};
  const keys = (filterQuery || []).map((filter) => filter.key);
  if (Array.isArray(filterIds)) {
    // return predefined list
    selector._id = { $in: filterIds };
  } else if (keys.length > 0) {
    // return filters that are part of the filterQuery
    selector.key = { $in: keys };
  } else {
    // do not return filters
    return null;
  }
  if (!includeInactive) {
    // include only active filters
    selector.isActive = true;
  }
  return selector;
};
```

So you could do the following to enable a global filter when searching:
```javascript
async transformFilterSelector(last) {
        // eslint-disable-line
        const { searchQuery = {} } = params;
        const { queryString, filterIds } = searchQuery;

        if (queryString && !filterIds) {
          return {
            ...(last || {}),
            _id: {
              $in: [
                GLOBAL_FILTER_ID,
              ],
            },
          };
        }

        return last;
      },
```
